### DISCUSSION TITLE
VDO_5WK96132Z MAF Data

### DISCUSSION THREAD

**Discussion by DeeEmm** [DIY-Flow-Bench/DIY-Flow-Bench#141] [General]
Original Source - https://www.ms4x.net/index.php?title=Aftermarket_Upgrade_Sensor_Data and also https://www.e46fanatics.com/threads/audi-rs4-maf-on-siemens-ms43-bosch-0-280-218-067.1113160/

```
MAF Type: Voltage (0-5v)
Original housing diameter: 72mm
Original Units: KG/H
Part #: 5WK96132Z / 13627567451
OEM Application: BMW M54B30
Status: UNTESTED
Max Flow: 1023 kg/h (502 cfm)
```

### MAF Data

<details><summary>VDO 5WK96132Z data (click to expand)</summary>

millivolts | kg/hour
-- | --
0 | 0
5 | 1.78
10 | 2.23
20 | 2.69
40 | 3.14
60 | 3.58
80 | 4.03
100 | 4.48
120 | 4.92
140 | 5.38
160 | 5.81
180 | 6.28
200 | 6.72
220 | 7.25
240 | 7.55
260 | 7.86
280 | 8.17
300 | 8.48
320 | 8.81
340 | 9.14
360 | 9.47
380 | 9.81
400 | 10.17
420 | 10.53
440 | 10.89
460 | 11.27
480 | 11.66
500 | 12.05
520 | 12.45
540 | 12.88
560 | 13.3
580 | 13.73
600 | 14.19
620 | 14.64
640 | 15.09
660 | 15.58
680 | 16.06
700 | 16.56
720 | 17.06
740 | 17.58
760 | 18.11
780 | 18.64
800 | 19.19
820 | 19.75
840 | 20.31
860 | 20.91
880 | 21.5
900 | 22.09
920 | 22.72
940 | 23.34
960 | 24
980 | 24.66
1000 | 25.33
1020 | 26.02
1040 | 26.73
1060 | 27.45
1080 | 28.2
1100 | 28.95
1120 | 29.73
1140 | 30.53
1160 | 31.34
1180 | 32.19
1200 | 33.05
1220 | 33.94
1240 | 34.84
1260 | 35.77
1280 | 36.72
1300 | 37.69
1320 | 38.69
1340 | 39.69
1360 | 40.72
1380 | 41.77
1400 | 42.83
1420 | 43.91
1440 | 44.98
1460 | 46.09
1480 | 47.2
1500 | 48.33
1520 | 49.47
1540 | 50.61
1560 | 51.77
1580 | 52.92
1600 | 54.09
1620 | 55.28
1640 | 56.47
1660 | 57.67
1680 | 58.89
1700 | 60.13
1720 | 61.38
1740 | 62.64
1760 | 63.92
1780 | 65.22
1800 | 66.53
1820 | 67.88
1840 | 69.23
1860 | 70.63
1880 | 72.03
1900 | 73.45
1920 | 74.92
1940 | 76.41
1960 | 77.92
1980 | 79.47
2000 | 81.05
2020 | 82.64
2040 | 84.28
2060 | 85.97
2080 | 87.67
2100 | 89.42
2120 | 91.2
2140 | 93.03
2160 | 94.89
2180 | 96.78
2200 | 98.7
2220 | 100.67
2240 | 102.67
2260 | 104.7
2280 | 106.78
2300 | 108.88
2320 | 111.03
2340 | 113.2
2360 | 115.42
2380 | 117.67
2400 | 119.95
2420 | 122.28
2440 | 124.64
2460 | 127.03
2480 | 129.47
2500 | 131.94
2520 | 134.45
2540 | 137
2560 | 139.58
2580 | 142.2
2600 | 144.86
2620 | 147.56
2640 | 150.3
2660 | 153.08
2680 | 155.89
2700 | 158.75
2720 | 161.66
2740 | 164.59
2760 | 167.56
2780 | 170.58
2800 | 173.64
2820 | 176.75
2840 | 179.89
2860 | 183.09
2880 | 186.33
2900 | 189.63
2920 | 192.95
2940 | 196.34
2960 | 199.77
2980 | 203.25
3000 | 206.78
3020 | 210.36
3040 | 213.98
3060 | 217.67
3080 | 221.41
3100 | 225.19
3120 | 229.03
3140 | 232.92
3160 | 236.88
3180 | 240.89
3200 | 244.95
3220 | 249.09
3240 | 253.28
3260 | 257.52
3280 | 261.83
3300 | 266.2
3320 | 270.64
3340 | 275.14
3360 | 279.72
3380 | 284.34
3400 | 289.05
3420 | 293.83
3440 | 298.67
3460 | 303.59
3480 | 308.58
3500 | 313.66
3520 | 318.8
3540 | 324.03
3560 | 329.34
3580 | 334.73
3600 | 340.2
3620 | 345.78
3640 | 351.44
3660 | 357.17
3680 | 363.02
3700 | 368.95
3720 | 374.98
3740 | 381.13
3760 | 387.38
3780 | 393.72
3800 | 400.17
3820 | 406.75
3840 | 413.42
3860 | 420.23
3880 | 427.16
3900 | 434.2
3920 | 441.38
3940 | 448.67
3960 | 456.11
3980 | 463.69
4000 | 471.39
4020 | 479.27
4040 | 487.28
4060 | 495.45
4080 | 503.78
4100 | 512.28
4120 | 520.95
4140 | 529.8
4160 | 538.83
4180 | 548.06
4200 | 557.47
4220 | 567.09
4240 | 576.92
4260 | 586.95
4280 | 597.22
4300 | 607.7
4320 | 618.39
4340 | 629.25
4360 | 640.28
4380 | 651.47
4400 | 662.78
4420 | 674.2
4440 | 685.73
4460 | 697.34
4480 | 709.02
4500 | 720.75
4520 | 732.52
4540 | 744.31
4560 | 756.09
4580 | 767.89
4600 | 779.69
4620 | 791.48
4640 | 803.3
4660 | 815.13
4680 | 826.97
4700 | 838.86
4720 | 850.77
4740 | 862.72
4760 | 874.72
4780 | 886.78
4800 | 898.88
4820 | 911.05
4840 | 923.3
4860 | 935.61
4880 | 948
4900 | 960.48
4920 | 973.06
4940 | 985.73
4960 | 998.53
4980 | 1011.42
5000 | 1023.98

</details>

### Calculated Coefficients

Coefficient | Value
-- | --
C0 | -1.095527704
C1 | 0.059430983
C2 | -0.000100137
C3 | 9.78512E-08
C4 | -3.88608E-11
C5 | 7.41299E-15
C6 | -5.03733E-19

R2 value: 0.999912426

### Comparison Graph

<img width="950" alt="Screenshot 2025-02-06 at 11 53 35 pm" src="https://github.com/user-attachments/assets/0e957f32-6d16-4eb3-9a3f-f42ee1f92e59" />

**Comment by DeeEmm** [comment]
This MAF Data file supersedes the original file replacing the data measured on the flow bench with the published data sourced from the links above. There is some considerable difference between the published data and the original measured data which I suspect highlights inaccuracies with the method we used to calculate the MAF Data.

The original file used a three point method that took three orifices measured on a Superflow bench and then used the voltage mapped from measuring the same orifices with the MAF to extrapolate the data. Not only are there are a lot of environmental factors that can influence the results but I suspect that the transfer function that I created was inaccurate. These differences highlight the need for accurate baseline data and a repeatable calibration method.

**Comment by black-top** [comment]
So I made an adapter to mount second maf at my small pump flow bench. 
I used the same input and was switching between 5WK96132 and 0280218067. 
Trim pot value same -0.03, both mafs reads correctly with that, flow calibration offset 0 for both.

0280218067 with cd @10" flows 23scfm , "100cfm orfice @28" flows 135scfm.
5WK96132 with cd@10" flows 13scfm, "100cfm orifice @28" flows 97scfm. 

So 5WK96132 spot on, even without calibration. 

And I did test with same cylinder head, only problem switching beetwen mafs that esp32esp 32 needs to be restarted.
 Flow difference between mafs is huge. The flow with 5WK96132 I think is realistic. The question why 0280218067 flow is so much bigger?


![0280218067-iso1585at28](https://github.com/user-attachments/assets/1e0633fc-01a2-4845-bb49-1d885058e995)

![5WK96132-iso1585at28](https://github.com/user-attachments/assets/524d2e28-fdb7-445f-9bb6-bd39ea7ec4da)

**Reply by DeeEmm** [reply]
Each MAF will need to be calibrated individually

> 0280218067 with cd @10" flows 23scfm , "100cfm orfice @28" flows 135scfm.

There is already a 35cfm error at 100cfm. so at 200cfm this could be 70cfm and 84cfm @ 240cfm. Which is exactly what you are showing (looks something like 324cfm)

**Reply by black-top** [reply]
How current calibration algorithm works? If I calibrate with 100cfm orifice, it just subtracts 35cfm from everywhere?

**Reply by black-top** [reply]
I did some testing today, and as I expected with maf calibrated with a 100 cfm orifice at 28 I get correct flow for the orifice, but I get -24CFM at 10" with a CD. So I think calibration like this is not completely correct. Maybe something like this will be more appropriate. For example, without calibration, 0280218067 maf flow 135CFM @28 with 100cmf orifice, so 35CFM offset, so maybe if we divide offset by measured flow (135-100)/135 we get 0.2593, so about 26% error, so for example calibrated flow =  uncalibrated 135 * ( 1-0.2593)= 100CFM  and for CD 18*(1-0.2593)=13.3326CFM .  (18CFM is flow for CD @10" without calibration) 
Not exactly but close, and we get a closer result under all maf range in comparing just subtraction by 35CFm from all range.

**Reply by black-top** [reply]
0280218067 flow corrected by % error follows quite close to m54b30 MAF . And current calibration is correct only at a single point. 


![corected1](https://github.com/user-attachments/assets/db596480-2eb9-42eb-994e-ee8401de234c)

**Reply by DeeEmm** [reply]
> I get -24CFM at 10" with a CD. So I think calibration like this is not completely correct.

I think that the CD is not appropriate for a proper bench. It is okay for a shop-vac style bench, but once you start to pull a proper reference pressure they are not ideal.

This is why I was keen to use the saw blade as they have a larger diameter hole (up to 22mm IIRC) and they do not flex. The only issue is that we need to get them tested on a proper bench to give us a point of reference.

**Reply by DeeEmm** [reply]
Ideally the bench needs to be calibrated at the reference pressure that it will be used at. So for a bench that can flow 300cfm @ 28"/H2O, the calibration orifice should also be somewhere near to that value.  Trying to calibrate a bench at 14cfm and then expecting accuracy at the other end of the operating range is a tall order.

**Reply by black-top** [reply]
I can do the same test with saw blade too, but as You can see with 5WK96132 MAF I get really close results for CD and 100CFM orifice even without any calibration, only with the trim pot for ADC offset, but it is the same for both MAF. 
And you can see at the graph that with the current calibration method, just by substacktion flow offset will give correct readings only at that point, but if we calculate flow offset with % of error, we get results under all curves very close for both MAFS 0280218067  and 5WK96132.

**Reply by black-top** [reply]
And BTW, I adjust pressure for 100CFM orifice to 28" and for CD to 10" when doing those tests. I dont measure CD at 28".

**Reply by DeeEmm** [reply]
Can you explain the % error calculation to me.

The two traces for the RS4 MAF (blue and green) follow exactly the same characteristics. This is what I would expect when comparing results where one had been calibrated and one not as the calibration simply shifts the line in the vertical axis, otherwise they are identical.

The Yellow MAF appears to have a completely different characteristic. But this is not dissimilar to the Red trace (% correction.) So I would like to understand what the % correction consists of

**Reply by DeeEmm** [reply]
(I would expect the red line to follow the blue and green)

**Reply by black-top** [reply]
Sorry, English is not my first language, so it is difficult for me to explain everything clearly.  I have mounted two MAFs 5wk96132 and 0280218067 one after the other at the same flow bench and did flow testing at the same time, by switching MAF signal wires and changing MAF type at the settings.  
So the yellow line is 5wk96132 MAF, with that MAF I get 13.5-14CFM for CD @ 10" and about 98CFM for the "100CFM orifice." @ 28" without any flow calibration, so it is almost exactly what flow is expected to see.
Now when I do same test with 0280218067 MAF I get 24CFM @ 10" for CD and 135CFM for 100CFM orifice @28".  So we have 26% error for 100CFM orifice. 
The green line is 35CFM offset calibration. It just subtracts 35CFM from everywhere.
 
The red line is 0280218067 MAF with corrected 26% error for each lift point measurement, and you can see that it looks very similar to 5wk96132 measurements.

**Reply by black-top** [reply]
>For example, without calibration, 0280218067 maf flow 135CFM @28 with 100cmf orifice, so 35CFM offset, so maybe if we divide offset by measured flow (135-100)/135 we get 0.2593, so about 26% error, so for example calibrated flow = uncalibrated 135 * ( 1-0.2593)= 100CFM and for CD 18*(1-0.2593)=13.3326CFM . (18CFM is flow for CD @10" without calibration)

**Reply by black-top** [reply]
So I think is better to do calibration like this: measure the known orifice at 28" in this case, the 100CFM orifice. With 067 MAF I see 135CFM instead of 100CFM, so 26% error. Now instead of subtraction of 35CFM like is now done at the code, we use 26% error correction for flow at all points, so we get more accurate results at any flow, not just at the calibration point.

**Reply by DeeEmm** [reply]
Ahh yes I understand now.

Applying the offset as a percentage of the original value changes the characteristic. (the shape of the line) This was not my intent with my original comment, It was just to try and explain the magnitude of the difference between the two.

What I note is that the lines should cross at a common point at 100cfm as this is the calibration value and a value at which both sensors should be common, however they do not.

Both line should follow the same trajectory. I am not sure why they do not. 

> So I think is better to do calibration like this: measure the known orifice at 28" in this case, the 100CFM orifice. With 067 MAF I see 135CFM instead of 100CFM, so 26% error. Now instead of subtraction of 35CFM like is now done at the code, we use 26% error correction for flow at all points, so we get more accurate results at any flow, not just at the calibration point.

I am unsure on using a percentage offset.. The characteristics of the MAF sensors should be that they are accurate across their range. And apart from providing a fixed offset to make the initial correction, should all follow a similar plot trajectory. The main difference is that the larger sensors simply extend further.

hmmnn okay. Now I think I know what the issue may be...

The RS4 MAF is 82mm in diameter.
The BMW MAF is only 72mm in diameter 

Are you entering the diameter of the pipework?

**Reply by black-top** [reply]
Yes, but for example 0280218067 with CD @ 10"  I get 24CFm, with 26% error correction I get 17.7CFm and if I use 35 offset I get 24-35= -11CFM. It looks like it gives better results.  I use original housings for both mafs, so I think it should not be affected by the housing size ? I got the same results for 067 MAF before mounting an additional 5wk96132 MAF.

**Reply by black-top** [reply]
Well, I think it is logical if the error is consistent across all measurement range, it is proportional to flow, so I think % error correction should be correct.  At higher flows, the absolute error is larger, but it is still the same proportion of the flow

**Reply by DeeEmm** [reply]
> I use original housings for both mafs, so I think it should not be affected by the housing size

But the diameter of your system is then not matched to the MAF. This means that you will get turbulent flow from the junction where the diameter changes. This will have a direct impact on the flow reading. Tube should be same diameter as MAF for both sides of MAF sensor



> I think it is logical if the error is consistent across all measurement range, it is proportional to flow, so I think % error correction should be correct. At higher flows, the absolute error is larger, but it is still the same proportion of the flow

No. Not necessarily. This is an assumption based on a single observation. This test also has other issues as there is an additional issue introduced with the diameter change. We also do not know the provenance of the MAF data for the Audi sensor. It may be incorrect / inaccurate and the error observed may simply be as a result of bad MAF data.

Too many unknown variables to determine that a percentage correction is required. The application of this correction also does not then make the data from the 5wk96132 MAF match perfectly, There are still large deviations / errors, so there is something more fundamental going on here.

The flow characteristics of the MAF sensor should be correct across its range. Applying a percentage correction indicates that the error in MAF reading increases with flow. This would point to an error in the calculation, however we can compare the KG/H flow value for any given voltage against the MAF data for the Audi sensor and note that the deviation is less than 3 kg/h. So we know that the calculation itself is correct

I already noted elsewhere that there is a larger data spread with the Audi MAF, as I observed that the ideal polynomial representation of the data was not in line with the MAF data itself. As I mentioned, this indicates that there was potentially an issue with the MAF data. Comparatively data obtained from the Bosch Data sheets appeared to be much more in line with expectations.

I think that this does raise the question of the validity of the data for the Audi Sensor. There is every likelyhood that the data was created by the end user and is not from a valid source. Reading back through the forum thread that the data was posted in leads me to think that this is the case. The MAF data is reported to improve performance in several installations discussed in the thread, but this does not automatically mean that the data is accurate. It simply means that the ECU is delivering more fuel for any given load point than it did with the stock MAF sensor. So I would actually conclude from this that the MAF data is NOT accurate as it is intentionally fooling the ECU into thinking that it is flowing more air than it actually is.

The premise is that the Audi MAF sensor is used is to increase the air flow range. But if this was the case I would expect that everything under the original MAF sensors limit to be exactly the same and the line from t4h Audi MAF simply extends further. However this is not the case as can be seen from the image posted below

![curves](https://github.com/user-attachments/assets/b224ae51-4ced-4a1e-b6c7-3053911da674)


This is from the same thread that the MAF Data was taken from and is a comparison made by someone who installed the Audi MAF and modded their tune.

It is also worth noting that no one in thread is actually 'tuning' their cars. They are simply substituting the MAF sensor and MAF data. They are not installing the MAF, increasing the MAF range and then tuning the AFR table, which is what you would expect.

**Reply by DeeEmm** [reply]
I guess the next question then becomes how can we test this and validate the MAF readings??

We can calibrate to one data point - that of the calibration orifice. But I think that we also need a range of additional orifices to test accuracy at other data points.

So we calibrate at a chosen data point. i.e. Ideally approx 80% of published MAF flow. 
Then we test at various other data points along the scale - perhaps 10% increments.

This means that we need multiple calibrated orifices or a way to approximate them.

I did purchase this device some time ago... 

![PXL_20250212_234132715](https://github.com/user-attachments/assets/fc9ce095-39c1-46de-b07f-48a1ce284aad)

It's a drafting device for drawing circles. A quick way to approximate different orifices.

We could calculate the idea sharp edged orifice diameter for any given flow and reference pressure and then set this device to that diameter. This would allow a quick and easy validation test but would obviously not be as accurate as a proper manufactured and calibrated orifice. But I would expect that it could get to within a few CFM of a target. Enough to be able to validate the MAF to within a reasonable error.

**Reply by DeeEmm** [reply]
Thinking further on this. I think that we are now essentially at the data validation stage. I think that we should continue this discussion in that thread

https://github.com/DeeEmm/DIY-Flow-Bench/issues/35
